tap_shopify/streams/base.py

```diff
@@ -212,108 +222,107 @@ def update_bookmark(self, bookmark_value, bookmark_key=None):
-                    if replication_value > max_bookmark:
-                        max_bookmark = replication_value
+                    data = self.call_api(query_params)
+
+                for edge in data.get("edges"):
+                    obj = self.transform_object(edge.get("node"))
                     yield obj
 
-                # You know you're at the end when the current page has
-                # less than the request size limits you set.
-                if len(objects) < self.results_per_page:
-                    # Save the updated_at_max as our bookmark as we've synced all rows up in our
-                    # window and can move forward. Also remove the since_id because we want to
-                    # restart at 1.
-                    stream_bookmarks = Context.state.get('bookmarks', {}).get(self.name, {})
-                    stream_bookmarks.pop('since_id', None)
-                    stream_bookmarks.pop('updated_at_max', None)
-                    self.update_bookmark(utils.strftime(updated_at_max))
-                    break
-
-                if objects[-1].id != max([o.id for o in objects]):
-                    # This verifies the api behavior expectation we have
-                    # that all pages are internally ordered by the
-                    # `since_id`.
-                    raise OutOfOrderIdsError("{} is not the max id in objects ({})".format(
-                        objects[-1].id, max([o.id for o in objects])))
-                since_id = objects[-1].id
-
-                # Put since_id and updated_at_max into the state.
-                self.update_bookmark(since_id, bookmark_key='since_id')
-                self.update_bookmark(utils.strftime(updated_at_max), bookmark_key='updated_at_max')
-
-            updated_at_min = updated_at_max
-        bookmark = max(min(stop_time,
-                           max_bookmark),
-                       (stop_time - datetime.timedelta(days=date_window_size)))
-        self.update_bookmark(utils.strftime(bookmark))
+                page_info =  data.get("pageInfo")
+                cursor , has_next_page = page_info.get("endCursor"), page_info.get("hasNextPage")
 
-    def sync(self):
-        """Yield's processed SDK object dicts to the caller.
+            last_updated_at = query_end
 
-        This is the default implementation. Get's all of self's objects
-        and calls to_dict on them with no further processing.
+    def sync(self):
+        """
+        Default implementation for sync method
         """
+        start_time = utils.now().replace(microsecond=0)
+        max_bookmark_value = current_bookmark_value = self.get_bookmark()
+
         for obj in self.get_objects():
-            yield obj.to_dict()
+            replication_value = utils.strptime_to_utc(obj[self.replication_key])
+
+            max_bookmark_value = max(max_bookmark_value, replication_value)
+
+            if replication_value >= current_bookmark_value:
```

Comment:
Is this really required?

Reply:
It's usually required for the pseudo incremental sync.
I have moved the sync function to the respective class and made the sync function in the base file more generic.